From 2c22c5d12d1fe7b27685fe5e7a9b365f33d3c952 Mon Sep 17 00:00:00 2001
From: Lawrence Rust <lvr@softsystem.co.uk>
Date: Sun, 17 Jun 2012 10:30:37 +0200
Subject: [PATCH 37/56] libmyth: Prevent SEGV in avcodec_decode_video2 when
 playing nuv material

Under certain circumstances NuppelDecoder::DecodeFrame can call
avcodec_decode_video2 when mpa_vidctx is NULL and thus cause a SEGV.

This patch adds a test for a NULL mpa_vidctx exits with error.

Signed-off-by: Lawrence Rust <lvr@softsystem.co.uk>
Signed-off-by: Jean-Yves Avenard <jyavenard@mythtv.org>
---
 mythtv/libs/libmythtv/nuppeldecoder.cpp |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/mythtv/libs/libmythtv/nuppeldecoder.cpp b/mythtv/libs/libmythtv/nuppeldecoder.cpp
index 11b4e8b..4037e0c 100644
--- a/mythtv/libs/libmythtv/nuppeldecoder.cpp
+++ b/mythtv/libs/libmythtv/nuppeldecoder.cpp
@@ -906,6 +906,12 @@ bool NuppelDecoder::DecodeFrame(struct rtframeheader *frameheader,
         if (!mpa_vidcodec)
             InitAVCodecVideo(frameheader->comptype - '3');
 
+        if (!mpa_vidctx)
+        {
+            LOG(VB_PLAYBACK, LOG_ERR, LOC + "NULL mpa_vidctx");
+            return false;
+        }
+
         MythAVFrame mpa_pic;
         if (!mpa_pic)
             return false;
-- 
1.7.10.2

